eda/data_analysis.py

```diff
@@ -104,9 +122,82 @@ def prepare_data(df: pd.DataFrame) -> pd.DataFrame:
         DataFrame to be used t
 
     """
-    return df.groupby(["SpendMonth", "Category", "MarketSector"], as_index=False).agg(
-        {"EvidencedSpend": "sum"}
-    )
+    return df.groupby(["SpendMonth", "Category", "MarketSector"], as_index=False).agg({"EvidencedSpend": "sum"})
+
+
+def prepare_data(df: pd.DataFrame, category: str, sector: str) -> pd.DataFrame:
+    """
+    Generate a correct data range and index for the Dataframe to be used for training.
+    Args:
+        df: Input DataFrame
+        category: Category for which spend data should be fetched
+        sector: Market sector for which spend data should be fetched
+
+    Returns:
+        Generate the correct date range to be used to train models.
+
+    """
+    category_sector_spend = df[(df["Category"] == category) & (df["MarketSector"] == sector)].reset_index(drop=True)
+    category_sector_spend.set_index("SpendMonth", inplace=True, drop=False)
+    # complete_data = pd.DataFrame(data=pd.date_range(start=category_sector_spend["SpendMonth"].min(axis=0),
+    #                                                 end=category_sector_spend["SpendMonth"].max(axis=0), freq="MS"),
+    #                              columns=["SpendMonth"])
```

Comment:
can we remove this commented out code if it is not needed?

Reply:
Good catch. Fixed!